inc/classes/class-admin-settings.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * Admin Settings Page class
+ *
+ * @package rt-pwa-extensions
+ */
+
+namespace RT\PWA\Inc;
+
+use \RT\PWA\Inc\Traits\Singleton;
+
+/**
+ * Class Admin_Settings
+ */
+class Admin_Settings {
+	use Singleton;
+
+	/**
+	 * @var string settings page slug.
+	 */
+	private $page_slug = 'pwa_extension';
+
+	/**
+	 * Admin_Settings constructor.
+	 */
+	protected function __construct() {
+		// Setup hooks.
+		add_action( 'admin_init', array( $this, 'register_plugin_settings' ) );
+		add_action( 'admin_menu', array( $this, 'register_options_page' ) );
+	}
+
+	/**
+	 * Register plugin setting.
+	 */
+	public function register_plugin_settings() {
+		register_setting( $this->page_slug, 'rt_pwa_extension_options', array( $this, 'validate_setting' ) );
```

Comment:
https://github.com/rtCamp/pwa-extension/pull/18#discussion_r364102084

This feedback is not addressed.